### DISCUSSION TITLE
CIR in directional beam scenario using ray-tracing

### DISCUSSION THREAD

**Discussion by Advaith2212** [NVlabs/sionna#234] [Q&A]
Hi,

I would like to generate CIR where TX is transmitting steered beam towards user using ray-tracing.

```py
sionna.rt.Scene.coverage_map(self, rx_orientation=(0.0, 0.0, 0.0), max_depth=3, cm_center=None, cm_orientation=None, cm_size=None, cm_cell_size=(10.0, 10.0), combining_vec=None, precoding_vec=None, num_samples=2000000, los=True, reflection=True, diffraction=False, scattering=False, edge_diffraction=False, check_scene=True)
```

Having combing_vec and steering_vec as input, can we get path and CIR from coverage map object ?

Thanks and regards,
Advaith

**Comment by jhoydis** [comment]
Hi Advaith,

If you want to get channel impulse responses (CIR), you need to use the ``scene.compute_paths`` function. The coverage map does not compute channel impulse responses explicitly.

**Reply by Advaith2212** [reply]
Oh ok sir,

Is it possible to obtain CIR from steered beams ?

**Reply by jhoydis** [reply]
Yes. The workflow for this is that you need to generate CIRs while having a TX array with multiple antennas configured. Then you apply to the CIRs the desired steering/precoding vector.

**Reply by NaveenMP** [reply]
Are 3GPP beamforming codebooks supported in Sionna?
